docs/usage.rst

```diff
@@ -0,0 +1,257 @@
+        hop_length = int(hop_dur * sampling_rate)
+        win_length = int(win_dur * sampling_rate)
+        mfcc = librosa.feature.mfcc(
+            y=signal,
+            sr=sampling_rate,
+            n_mfcc=13,
+            hop_length=hop_length,
+            win_length=win_length,
+        )
+        return mfcc
+
+    win_dur = 0.02
+    hop_dur = 0.01
+    n_mfcc = 13
+    interface = audinterface.Feature(
+        [f'mfcc-{idx}' for idx in range(n_mfcc)],
+        process_func=features,
+        process_func_args={
+            'win_dur': win_dur,
+            'hop_dur': hop_dur,
+            'n_mfcc': n_mfcc,
+        },
+        hop_dur=hop_dur,
+        win_dur=win_dur,
+    )
+    df = interface.process_index(index)
+    df
+
+
+Serializable feature interface
+------------------------------
+
+To use a feature extractor as an input transform
+of a machine learning model
+it is recommend to provide it in a serializable way
+so it can be `stored as part of the model`_.
+One example of such a feature extractor is :class:`opensmile.Smile`.
+
+To create such a feature extractor,
+we create a class that inherits
+from :class:`audinterface.Feature`
+and :class:`audobject.Object`.
+
+.. jupyter-execute::
+
+    import audobject
+
+    class MeanStd(audinterface.Feature, audobject.Object):
+
+        def __init__(self):
+            super().__init__(
+                ['mean', 'std'],
+                process_func=self.features,
+            )
+
+        def features(self, signal, sampling_rate):
+            return [signal.mean(), signal.std()]
+
+    fex = MeanStd()
+    df = fex.process_index(index)
+    df
+
+The advantage of the feature extraction object is
+that we can save it to a YAML file
+and re-instantiate it from there.
+
+.. jupyter-execute::
+
+    fex.to_yaml('mean-std.yaml')
+    fex2 = audobject.from_yaml('mean-std.yaml')
+    df = fex2.process_index(index)
+    df
+
+
+Segmentation interface
+----------------------
+
+.. jupyter-execute::
+
+    import auditok
+
+    def segments(signal, sampling_rate):
+
+        # Convert floating point array to 16bit PCM little-endian
+        ints = (signal[0, :] * 32767).astype(np.int16)
+        little_endian = ints.astype('<u2')
+        signal = little_endian.tobytes()
+
+        regions = auditok.split(
+            signal,
+            sampling_rate=sampling_rate,
+            sample_width=2,
+            channels=1,
+            min_dur=0.2,
+            energy_threshold=70,
+        )
+        index = pd.MultiIndex.from_tuples(
+            [
+                (
+                    pd.Timedelta(region.meta.start, unit='s'),
+                    pd.Timedelta(region.meta.end, unit='s'),
+                )
+                for region in regions
+            ],
+            names=['start', 'end'],
+        )
+        return index
+
+    interafce = audinterface.Segment(process_func=segments)
+    idx = interafce.process_file(files[0])
```

Comment:
```suggestion
    idx = interface.process_file(files[0])
```

Reply:
I think I would also rename to `index`, with `idx` I associate an integer.

Reply:
Ah wait, we already have a variable `index`.

Reply:
Actually, is there a specific reason that you assign the result and not just show the output directly (applies also to the other examples).

Reply:
The reason behind is that the user can more easily see what type the method returns.
If we just show the HTML formated output I'm afraid that it is not that easy to differentiate between Series, DataFrame, and Index

Reply:
ok

Reply:
And yes, I used `idx` instead of `index` to not overwrite the already assigned `index` variable.

We could call it `segmented_index`, but I was also not convinced to add an additional word as we also don't do it for the other examples.

Reply:
Yes, I guess there is no perfect solution, so let's stick with `idx`.